packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -62,16 +62,30 @@ cortexProxy.on("proxyRes", (proxyRes, req, res) => {
        * - https://github.com/cortexproject/cortex/blob/82b32ec65ed16920e6053ac7fb748c42e3cae452/pkg/compactor/compactor_http.go#L16-L25
        */
       const body = chunks.join();
-      const parsedHTML = parse(body);
-      const errorMessage = parsedHTML.querySelector("html body p")?.text;
-
-      /* setting the status code as 412, as usually HTML error responses
-       * of cortex are due to misconfigurations
-       */
-      res.writeHead(errorMessage ? 412 : 500, {
-        "content-type": "text/plain"
-      });
-      res.end(errorMessage ?? "An unknown error occured.");
+      try {
+        const parsedHTML = parse(body);
+        const errorMessage = parsedHTML.querySelector("html body p")?.text;
+
+        if (!errorMessage) {
+          throw new Error("Could not extract error message");
+        }
+
+        /* setting the status code as 412, as usually HTML error responses
+         * of cortex are due to misconfigurations
+         */
+        res.writeHead(412, {
+          "content-type": "text/plain"
+        });
+        res.end(errorMessage);
+      } catch (e) {
+        res.writeHead(500, {
+          "content-type": "text/plain"
+        });
+        res.end(
+          "There has been an error parsing a cortex HTML response. Response:\n\n" +
+            body
```

Comment:
Also food for thought for you to learn from the mess we've experienced in the past: so -- this here is _specifically_ for handling an _unexpected_ HTTP response. In which ways can an HTTP response be unexpected? In many ways. In all of them. And this piece of code _can_ handle all of them, it's actually pretty easy. There's just one thing to keep in mind: sometimes, an unexpected HTTP response may contain a _wild_ response body full of gibberish. This can be 1 byte or a gigabyte. In those cases it makes sense to emit only first N bytes.

Reply:
makes sense - though I shortened it to 500 characters instead of 500 bytes.